ci/compiled_size.py

```
@@ -141,21 +142,117 @@ def _run_pio_size(build_dir: Path) -> int | None:
+            # prog_path may end in .bin (what fbuild emits as `prog_path`)
+            # or .elf (what `_override_prog_path_for_fbuild` writes). The
+            # size tool only takes ELF, so normalise to .elf.
+            elf_candidate = prog_path.with_suffix(".elf")
+            if elf_candidate.exists():
+                return elf_candidate
+
+    fbuild_root = build_dir / ".fbuild" / "build"
+    if not fbuild_root.is_dir():
+        return None
+
+    env_name = list(board_info.keys())[0] if isinstance(board_info, dict) else None
```

Ruff (0.15.17)

[warning] Prefer `next(iter(board_info.keys()))` over single element slice

Replace with `next(iter(board_info.keys()))`

(RUF015)